test/sidebar-additional-tabs.html

```diff
@@ -0,0 +1,120 @@
+                }
+
+                .inputs input {
+                    margin-bottom: 10px;
+                }
+            </style>
+        <link rel="stylesheet" type="text/css" href="../dev/en-US/sidebar.css" />
+        <script src='https://unpkg.com/react/umd/react.development.js'></script>
+        <script src="../dev/en-US/sidebar.js"></script>
+        </head>
+        <body>
+            <div class="sidebar">
+                <div class="inputs">
+                    <div>
+                        <label>
+                            <div>File Id</div>
+                            <input class="file" type="text" placeholder="Enter file id" />
+                        </label>
+                        <label>
+                            <div>Auth Token</div>
+                            <input class="token" type="text" placeholder="Enter auth token" />
+                        </label>
+                    </div>
+                    <button type="button" onclick="load(true)">Submit</button>
+                </div>
+                <div class="preview sidebar"></div>
+            </div>
+            <script>
+                const Tab = function(contents) {
+                    return React.createElement('div', null, contents);
+                }
+                function load(refreshOnComplete = false) {
+                    const { ContentSidebar } = Box;
+                    const token = document.querySelector('.token').value || localStorage.getItem('token');
+                    const fileId = document.querySelector('.file').value || localStorage.getItem('file');
+
+                    if (!token || !fileId) {
+                        return;
+                    }
+
+                    localStorage.setItem('token', token);
+                    document.querySelector('.token').value = token;
+
+                    localStorage.setItem('file', fileId);
+                    document.querySelector('.file').value = fileId;
+
+                    const placeholder = document.createElement('div');
+                    placeholder.textContent = 'test';
+
+                    const sidebar = new ContentSidebar();
+                    sidebar.show(fileId, token, {
+                        hasAdditionalTabs: true, 
+                        additionalTabs: 
+                            [
+                                {
+                                    id: 200,
+                                    title: 'Open with Gmail', // used as tooltip
+                                    element: Tab('gmail'), //<RecommendedAppTab />,
+                                    callback: () => window.alert('gmail'),
+                                    status: 'ADDED',
+                                },
+                                {
+                                    id: 201,
+                                    title: 'Open with Slack',
+                                    element: Tab('slack'), //<RecommendedAppTab />,
```

Comment:
Not sure if we should allow insertion of React.Nodes from the caller app. Maybe we keep a registry/map of the apps we want to be showing and just extract it out from there. Similar to how we do our icons.

Reply:
👍 Going to switch to imageURLs for now, if there's significant demand from platform customers for custom markup here we can reconsider nodes.

Reply:
@ctuckz this will change our data structure a bit, instead of passing in a react component we can just pass in the URL now.

Reply:
How are we going to handle resin tags if we're just passing in the URL? Is putting the tags in the ui element an option, or do they need to be in EUA?

Reply:
There's a way to programatically fire resin events via js as "clicks". We can do this in the callback.

If that doesn't work for you we can add an "interactionTarget" prop, and included that on the wrapper component that's in the sidebar.

Reply:
I didn't realize we could do it programmatically - that should work fine